runtime/backtrace_byt.c

```diff
@@ -103,10 +103,16 @@ static struct debug_info *find_debug_info(code_t pc)
 
 static int cmp_ev_info(const void *a, const void *b)
 {
-  code_t pc_a = ((const struct ev_info*)a)->ev_pc;
-  code_t pc_b = ((const struct ev_info*)b)->ev_pc;
+  const struct ev_info* ev_a = a;
+  const struct ev_info* ev_b = b;
+  code_t pc_a = ev_a->ev_pc;
+  code_t pc_b = ev_b->ev_pc;
+  int lnum_a = ev_a->ev_lnum;
+  int lnum_b = ev_b->ev_lnum;
   if (pc_a > pc_b) return 1;
   if (pc_a < pc_b) return -1;
+  if (lnum_a < lnum_b) return 1;
+  if (lnum_a > lnum_b) return -1;
```

Comment:
For pcs you return `1` if `pc_a` is bigger, but for line numbers you return `1` if `lnum_a` is smaller. What's the reason for doing things differently?

Reply:
Just an arbitrary choice from when I was experimenting. I pushed a revised commit with a full lexicographic comparison (I don't compare the filenames though because I don't think it is possible for different filenames to flow here).